> I mean, did you actually get some error because of those? The one at line 113 would be very surprising as `buffer` is assigned right after the variable declarations.

You are right about line 113, but I encountered the following errors during compilation:

/home/fk/Dokumente/workspace/modular_nrf/zephyr/subsys/modem/modem_stats.c:41:16: error: 'first' may be used uninitialized [-Werror=maybe-uninitialized]
/home/fk/Dokumente/workspace/modular_nrf/zephyr/subsys/modem/modem_stats.c:52:16: error: 'next' may be used uninitialized [-Werror=maybe-uninitialized]

These errors were produced using arm-zephyr-eabi-gcc targeting a Cortex-M33 MCU, with the -Werror flag treating warnings as errors. Given that Zephyr supports many different toolchains and optimization settings, I believe it's safer to explicitly initialize these variables to NULL to prevent similar warnings for other developers using different compiler settings.

--- a/subsys/modem/modem_stats.c
+++ b/subsys/modem/modem_stats.c
@@ -32,7 +32,7 @@ static void stats_buffer_list_append(struct modem_stats_buffer *buffer)
 
 static struct modem_stats_buffer *stats_buffer_list_first(void)
 {
-	struct modem_stats_buffer *first;
+	struct modem_stats_buffer *first = NULL;
 
 	K_SPINLOCK(&stats_buffer_lock) {
 		first = stats_buffer_from_node(sys_slist_peek_head(&stats_buffer_list));
@@ -43,7 +43,7 @@ static struct modem_stats_buffer *stats_buffer_list_first(void)
 
 static struct modem_stats_buffer *stats_buffer_list_next(struct modem_stats_buffer *buffer)
 {
-	struct modem_stats_buffer *next;
+	struct modem_stats_buffer *next = NULL;
 
 	K_SPINLOCK(&stats_buffer_lock) {
 		next = stats_buffer_from_node(sys_slist_peek_next(&buffer->node));

